### ISSUE TITLE
[BUG] Slow reaction times on machine with large core-count

### ISSUE THREAD

**Issue by lbhm** [aristocratos/bpytop#289]
**Describe the bug**

I recently got access to a dual-socket AMD Epyc 7742 machine with a total of 256 logical cores. One of the first things I did was to install bpytop to monitor system behavior. However, I noticed that bpytop reacts very slowly on this machine. It takes ~10 second for the clock on the top center to update even though I configured an update interval of 1s. Since I never had such a problem on my own machine, my first assumption was that this is somehow related to the high core-count.

**To Reproduce**

* Start bpytop
* The loading screen takes about 10 seconds
* After the startup sequence, an update also takes about ten seconds.

**Expected behavior**

I would expect bpytop to update after the configured interval (1s).

**Screenshots**

I can record a video of the slow update process if my description was not precise enough.

**Info (please complete the following information):**
 - bpytop version: 1.0.63
 - psutil version: 5.8.0
 - (Linux) Linux distribution and version: Ubuntu 20.04.1 LTS
 - Terminal used: xfce4-terminal 0.8.10
 - Font used: Monospace Regular
 - Python version: Python 3.8.5

**Additional context**

contents of `~/.config/bpytop/error.log`:
```
12/04/21 (17:09:58) | INFO: New instance of bpytop version 1.0.63 started with pid 1847342
12/04/21 (17:09:58) | INFO: Loglevel set to DEBUG
12/04/21 (17:09:58) | DEBUG: Using psutil version 5.8.0
12/04/21 (17:09:58) | DEBUG: CMD: /home/lennart-ldap/.local/bin/bpytop --debug
12/04/21 (17:10:11) | DEBUG: Collect and draw completed in 10.820735 seconds
12/04/21 (17:10:12) | DEBUG: Init completed in 13.047520 seconds
12/04/21 (17:10:54) | INFO: Exiting. Runtime 0:00:56
```

**Comment by aristocratos** [comment]
@lbhm 
You should git clone https://github.com/giampaolo/psutil.git
And then from that directory run `./scripts/internal/print_api_speed.py`

All the results should be sub 1 second and the max for any result should be 0.2-0.3 seconds.
That should tell you if there is a problem in the psutil api.

I don't think there is anything that could halt it as much as 10 seconds in the bpytop code. With 16 cores, the processing for the Core graphs should be around 20-30ms and a 16x increase in number of cores should still be far below 1 second.

**Comment by lbhm** [comment]
Hey @aristocratos , 

I just ran the script and indeed the `cpu_freq` call took 5.11 seconds. Thank you for the hint, I will reach out to the psutil dev.

**Comment by aristocratos** [comment]
@lbhm 
I can add an option to disable the CPU frequency check in the next update, since that particular function seems to be consistently slow on all systems I've tested, not 5 seconds slow, but usually a lot slower than any other function.
Should be a workaround for your problem then too.

Reopening this, otherwise I'm likely to forget :)

**Comment by jeanlst** [comment]
Hey @aristocratos,
I'm also facing the same thing. My `print_api_speed` call returned the following:

```
SYSTEM APIS               SECONDS
----------------------------------
sensors_battery           0.000015
cpu_count                 0.000018
disk_usage                0.000021
users                     0.000031
pid_exists                0.000040
virtual_memory            0.000067
swap_memory               0.000106
cpu_times                 0.000106
cpu_percent               0.000110
cpu_times_percent         0.000113
boot_time                 0.000141
cpu_stats                 0.000165
net_io_counters           0.000251
net_if_stats              0.000252
disk_io_counters          0.000311
disk_partitions           0.000389
net_if_addrs              0.000417
sensors_fans              0.000911
pids                      0.001042
cpu_count (cores)         0.002235
net_connections           0.016278
sensors_temperatures      0.018550
process_iter (all)        0.020328
cpu_freq                  1.278670

PROCESS APIS              SECONDS
----------------------------------
create_time               0.000001
nice                      0.000005
cwd                       0.000008
exe                       0.000009
num_fds                   0.000010
ionice                    0.000013
cpu_affinity              0.000015
status                    0.000016
memory_info               0.000020
num_threads               0.000020
cpu_times                 0.000020
ppid                      0.000021
name                      0.000021
io_counters               0.000023
uids                      0.000025
num_ctx_switches          0.000025
memory_percent            0.000026
is_running                0.000030
threads                   0.000035
gids                      0.000038
open_files                0.000038
cpu_percent               0.000049
cmdline                   0.000051
connections               0.000064
environ                   0.000064
cpu_num                   0.000115
username                  0.000189
terminal                  0.000651
memory_full_info          0.000758
memory_maps               0.002126
children                  0.009388
```

The clock (and the rest of the terminal) takes around 2 seconds to update every time, which makes it unusable since every time I press arrowdown for instance, it takes 2 seconds to go to the next process. My machine has 64 logical cores.

**Comment by aristocratos** [comment]
@lbhm @jeanlst 
This seems to have been fixed in psutil version 5.8.1, it's not released yet but you can get by cloning psutil and from the directory running:
`make setup-dev-env`
`make build`
`pip3 install .`

**Comment by aristocratos** [comment]
@lbhm @jeanlst
Added an option to toggle CPU frequency and a fix for some weird behaviour in the latest psutil in bpytop v1.0.64
